### PR TITLE
chore(eslint-plugin-compass): adds a rule to eslint-plugin-compass to enforce links in repo to have utm_source and utm_medium params

### PR DESCRIPTION
**Pull request by himanshusinghs**
## Description
In this PR we added an ESLint rule that will enforce links written in compass to always have the following UTM parameters:
- `utm_source=compass`
- `utm_medium=product`

To make it easier for reviewer the PR is split in two parts:
- ESLint rule related changes (this PR)
- Follow up changes in compass to abide by this rule (#4448)
 

### Checklist
- [ ] New tests and/or benchmarks are included
- [ ] Documentation is changed or added
- [ ] I have signed the MongoDB Contributor License Agreement (https://www.mongodb.com/legal/contributor-agreement)

## Motivation and Context

- [ ] Bugfix
- [ ] New feature
- [ ] Dependency update
- [x] Misc

## Types of changes

- [ ] *Backport Needed*
- [ ] Patch (non-breaking change which fixes an issue)
- [ ] Minor (non-breaking change which adds functionality)
- [ ] Major (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: configs/eslint-plugin-compass/rules/no-mongodb-link-without-utm-params.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    fixable: 'code',
+    hasSuggestions: true,
+  },
+
+  create(context) {
+    const isTestFile = /\.(test|spec)\.(js|jsx|ts|tsx)$/.test(
+      context.getPhysicalFilename()
+    );
+    return {
+      Literal(node) {
```
Comment (gribnoysup) [inline configs/eslint-plugin-compass/rules/no-mongodb-link-without-utm-params.js:26]: Would be nice also make it work with `TemplateLiteral` type

**Comment by gribnoysup** [comment]
> is it better to report not one but two errors for missing or differently configured utm_source and utm_medium

I don't think it's needed to split those in two: we always want both utm attrs to be present

> is the behaviour of the rule which is to always enforce utm_source=compass and utm_medium=product is fine or not. Because it means that even if we want to write a different utm_source/utm_medium it will report an invalidation which ofcource can be suspended with the eslint disable but might be annoying?

I don't see a use-case where we would want that so probably not needed. Maybe if this becomes shared with vscode / mongosh? But for now I wouldn't worry about it

File: configs/eslint-plugin-compass/rules/no-mongodb-link-without-utm-params.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        'Links pointing to *.mongodb.com must have utm_source=compass and utm_medium=product in query params',
+    },
+    fixable: 'code',
+    hasSuggestions: true,
+  },
+
+  create(context) {
+    const isTestFile = /\.(test|spec)\.(js|jsx|ts|tsx)$/.test(
+      context.getPhysicalFilename()
+    );
```
[the remark this answers is not in the document]
Comment (himanshusinghs) [reply configs/eslint-plugin-compass/rules/no-mongodb-link-without-utm-params.js]: Thank you, I just moved this to the config.

File: configs/eslint-plugin-compass/rules/no-mongodb-link-without-utm-params.js (hunk 1 above)
Comment (himanshusinghs) [reply configs/eslint-plugin-compass/rules/no-mongodb-link-without-utm-params.js:69]: This was pretty nice to look into 🤩 I added this as well in my recent commits

**Comment by himanshusinghs** [comment]
Closing this in favor of the linked PR

**Pull request closed without merging**